pkg/values/map.go

```diff
@@ -73,10 +74,34 @@ func (m *Map) UnwrapTo(toStruct any) error {
 	return d.Decode(m.Underlying)
 }
 
+func mapValueToMap(f reflect.Type, t reflect.Type, data any) (any, error) {
+	if f != reflect.TypeOf(map[string]Value{}) {
+		return data, nil
+	}
+
+	switch t {
+	case reflect.TypeOf(map[string]any{}):
+		dv := data.(map[string]Value)
+		d := map[string]any{}
+		for k, v := range dv {
+			unw, err := Unwrap(v)
+			if err != nil {
+				return nil, err
+			}
+
+			d[k] = unw
+		}
+
+		return d, nil
+	}
+	return data, nil
```

Comment:
Shouldn't this be an error? When does returning back unmodified data make sense?

Reply:
An error means that this decode hook should have handled the transformation but couldn't; returning the data unmodified means we fallback to the next decode hook and/or whatever other decode logic mapstructure uses by default.